server.go

```diff
@@ -106,7 +107,15 @@ func (s *Server) loop() {
 		}
 		s := newSocket(conn, s.baseHandler)
 		go func(s *socket) {
-			s.loop()
+			err := s.loop()
+			conn.Close() // doesnt hurt to add this here, and also forces close on exiting early, like a timeout
+			switch err.Error() {
+			case "timeout":
```

Comment:
If you're switching on the error text body, use global vars instead for consistency (or better still, switch on the `err` itself by creating global errors)